### PR TITLE
Remove duplicate SnapshotConfig validation code

### PR DESCRIPTION
**Pull request by steviez**
#### Summary of Changes
Use the is_snapshot_config_valid() helper function instead of repeated logic.

### CODE REVIEW

File: local-cluster/tests/common/mod.rs [hunk 1]
```diff
@@ -478,20 +478,12 @@ impl SnapshotValidatorConfig {
         accounts_hash_interval_slots: Slot,
         num_account_paths: usize,
     ) -> SnapshotValidatorConfig {
+        // Interval values must be nonzero
         assert!(accounts_hash_interval_slots > 0);
         assert!(full_snapshot_archive_interval_slots > 0);
+        assert!(incremental_snapshot_archive_interval_slots > 0);
```
Comment (steviez) [inline local-cluster/tests/common/mod.rs:484]: For consistency between `solana-validator` and local-cluster tests (or really any other callers), we could move the checks these assertions perform into `is_snapshot_config()` valid.

I don't feel too strongly on that point - I care more about replacing the stuff I'm ripping out with the helper

**Review by apfitzge — APPROVED** [review]
Change looks good to me. It appears that `is_snapshot_config_valid` has more strict logic that what was here before, but that's probably reasonable.

A couple of thoughts:
1. `is_snapshot_config_valid` uses `Slot::MAX` instead of `DISABLED_SNAPSHOT_ARCHIVE_INTERVAL`
2. why is `is_snapshot_config_valid` in validator instead of snapshot_config?

**Comment by steviez** [comment]
> 1. `is_snapshot_config_valid` uses `Slot::MAX` instead of `DISABLED_SNAPSHOT_ARCHIVE_INTERVAL`

Are you potentially looking at your local branch that isn't at tip? here is the tip:
https://github.com/solana-labs/solana/blob/13aff74f823357f808723205c8f0802e815f8770/core/src/validator.rs#L2259-L2284

> 2. why is `is_snapshot_config_valid` in validator instead of snapshot_config?

Good point - could shift it to a method on `SnapshotConfig`

**Comment by apfitzge** [comment]
> Are you potentially looking at your local branch that isn't at tip? here is the tip:

🤯 woops. Yeah you're right.

**Comment by steviez** [comment]
> Historically the only snapshot config validation was done in validator, then slowly added into other places. Also, IIRC, there was divergence on what was valid, depending on validator vs local-cluster tests/etc.

Got it, thanks for the context. I think it is god to have the validation aligned; if there is a strong reason for something in local-cluster to be different, I'm open to it. Otherwise, I think it is better if our test path mirrors the real path as close as possible.

> Probably fine to make this a method on SnapshotConfig. And probably fine to do that in a subsequent PR :)

🤝 - Maybe next I'm waiting on a ledger-tool run

**Pull request merged**